File: src/main/java/org/hwabeag/playerfly/commands/ItemCommand.java
package org.hwabeag.playerfly.commands;

import java.util.ArrayList;
import java.util.Objects;
import org.bukkit.Bukkit;
import org.bukkit.ChatColor;
import org.bukkit.Material;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.configuration.file.FileConfiguration;
import org.bukkit.entity.Player;
import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.meta.ItemMeta;
import org.hwabeag.playerfly.config.ConfigManager;
import org.jetbrains.annotations.NotNull;

public class ItemCommand implements CommandExecutor {
    FileConfiguration PlayerFlyConfig = ConfigManager.getConfig("playerfly");
    String Prefix;

    public ItemCommand() {
        this.Prefix = ChatColor.translateAlternateColorCodes('&', (String)Objects.requireNonNull(this.PlayerFlyConfig.getString("playerfly.prefix")));
    }

    public boolean onCommand(@NotNull CommandSender sender, @NotNull Command command, @NotNull String label, @NotNull String[] args) {
        if (sender instanceof Player p) {
            if (p.isOp()) {
                if (args.length == 0) {
                    p.sendMessage(this.Prefix + " /플라이권 지급 [닉네임] [분] - 유저에게 플라이권을 지급합니다.");
                    p.sendMessage(this.Prefix + " /플라이권 불러오기 [분] - 플라이권을 불러옵니다.");
                    return true;
                } else if (args[0].equalsIgnoreCase("불러오기")) {
                    ItemStack item = new ItemStack(Material.BOOK, 1, (short)3);
                    ItemMeta meta = item.getItemMeta();
                    meta.setDisplayName(ChatColor.translateAlternateColorCodes('&', "&a&l[플라이권]"));
                    ArrayList<String> loreList = new ArrayList();
                    loreList.add(ChatColor.translateAlternateColorCodes('&', "&f&l- 사용시 플라이 &a&l" + args[1] + " &f분 추가"));
                    loreList.add(ChatColor.translateAlternateColorCodes('&', "&f&l- &a&l땅에 클릭시 사용됩니다."));
                    meta.setLore(loreList);
                    item.setItemMeta(meta);
                    p.getInventory().addItem(new ItemStack[]{item});
                    p.sendMessage(this.Prefix + " 플라이권을 불러왔습니다.");
                    return true;
                } else {
                    if (args[0].equalsIgnoreCase("지급")) {
                        if (args.length == 1) {
                            p.sendMessage(this.Prefix + " /플라이권 지급 [닉네임] [분] - 유저에게 플라이권을 지급합니다.");
                            return true;
                        }

                        if (Bukkit.getServer().getPlayerExact(args[1]) != null) {
                            Player player = Bukkit.getServer().getPlayerExact(args[1]);
                            ItemStack item = new ItemStack(Material.BOOK, 1, (short)3);
                            ItemMeta meta = item.getItemMeta();
                            meta.setDisplayName(ChatColor.translateAlternateColorCodes('&', "&a&l[플라이권]"));
                            ArrayList<String> loreList = new ArrayList();
                            loreList.add(ChatColor.translateAlternateColorCodes('&', "&f&l- 사용시 플라이 &a&l" + args[2] + " &f분 추가"));
                            loreList.add(ChatColor.translateAlternateColorCodes('&', "&f&l- &a&l땅에 클릭시 사용됩니다."));
                            meta.setLore(loreList);
                            item.setItemMeta(meta);
                            player.getInventory().addItem(new ItemStack[]{item});
                            player.sendMessage(this.Prefix + " 플라이권을 불러왔습니다.");
                            return true;
                        }
                    }

                    p.sendMessage(this.Prefix + " /플라이권 지급 [닉네임] [분] - 유저에게 플라이권을 지급합니다.");
                    p.sendMessage(this.Prefix + " /플라이권 불러오기 [분] - 플라이권을 불러옵니다.");
                    return true;
                }
            } else {
                p.sendMessage(this.Prefix + " 당신은 권한이 없습니다.");
                return true;
            }
        } else {
            Bukkit.getConsoleSender().sendMessage(this.Prefix + " 인게임에서만 사용이 가능합니다.");
            return true;
        }
    }
}

File: src/main/kotlin/org/hwabeag/playerfly/commands/ItemCommand.kt
package org.hwabeag.playerfly.commands

import org.bukkit.Bukkit
import org.bukkit.Material
import org.bukkit.NamespacedKey
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player
import org.bukkit.inventory.ItemStack
import org.bukkit.persistence.PersistentDataType
import org.hwabeag.playerfly.PlayerFlyPlugin

class ItemCommand(private val plugin: PlayerFlyPlugin) : CommandExecutor {
    private val minuteKey = NamespacedKey(plugin, "fly_ticket_minutes")

    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<out String>): Boolean {
        if (!sender.isOp) {
            sender.sendMessage("${plugin.prefix()} 권한이 없습니다.")
            return true
        }

        if (args.isEmpty()) {
            sendHelp(sender)
            return true
        }

        return when (args[0].lowercase()) {
            "지급", "give" -> giveTicket(sender, args)
            else -> {
                sendHelp(sender)
                true
            }
        }
    }

    private fun giveTicket(sender: CommandSender, args: Array<out String>): Boolean {
        if (args.size < 2) {
            sender.sendMessage("${plugin.prefix()} /플라이권 지급 <닉네임> [분]")
            return true
        }

        val target = findOnlinePlayer(args[1]) ?: run {
            sender.sendMessage("${plugin.prefix()} 해당 플레이어가 접속 중이 아닙니다.")
            return true
        }

        val defaultMinutes = plugin.config.getInt("fly-ticket.default-minutes", 30)
        val minutes = when {
            args.size >= 3 -> args[2].toIntOrNull()
            else -> defaultMinutes
        } ?: run {
            sender.sendMessage("${plugin.prefix()} 분은 숫자로 입력해 주세요.")
            return true
        }

        if (minutes <= 0) {
            sender.sendMessage("${plugin.prefix()} 분은 1 이상이어야 합니다.")
            return true
        }

        target.inventory.addItem(createItem(minutes))
        sender.sendMessage("${plugin.prefix()} ${target.name}에게 ${minutes}분 플라이권을 지급했습니다.")
        target.sendMessage("${plugin.prefix()} ${minutes}분 플라이권을 받았습니다.")
        return true
    }

    private fun createItem(minutes: Int): ItemStack {
        val configuredMaterial = plugin.config.getString("fly-ticket.material", "BOOK").orEmpty()
        val material = Material.matchMaterial(configuredMaterial.uppercase()) ?: Material.BOOK
        val item = ItemStack(material)
        val meta = item.itemMeta ?: return item

        val displayName = plugin.config.getString("fly-ticket.display-name", "&a&l[플라이권]").orEmpty()
        val loreTemplate = plugin.config.getStringList("fly-ticket.lore")
        val lore = loreTemplate.map { line ->
            plugin.color(line.replace("%minutes%", minutes.toString()))
        }

        meta.setDisplayName(plugin.color(displayName))
        meta.lore = lore
        meta.persistentDataContainer.set(minuteKey, PersistentDataType.INTEGER, minutes)
        item.itemMeta = meta
        return item
    }

    private fun sendHelp(sender: CommandSender) {
        sender.sendMessage("${plugin.prefix()} /플라이권 지급 <닉네임> [분]")
    }

    private fun findOnlinePlayer(input: String): Player? {
        return Bukkit.getOnlinePlayers().firstOrNull { it.name.equals(input, ignoreCase = true) }
            ?: Bukkit.getPlayer(input)
    }
}
